overlord/ifacestate/handlers.go

```diff
@@ -763,3 +763,27 @@ func (m *InterfaceManager) undoTransitionUbuntuCore(t *state.Task, _ *tomb.Tomb)
 
 	return m.transitionConnectionsCoreMigration(st, newName, oldName)
 }
+
+// injectTasks makes all the halt tasks of the mainTask wait for extraTasks;
+// extraTasks join the same lane and change as the mainTask.
+func injectTasks(mainTask *state.Task, extraTasks *state.TaskSet) {
+	lanes := mainTask.Lanes()
+	if len(lanes) == 1 && lanes[0] == 0 {
+		lanes = nil
+	}
+	ht := mainTask.HaltTasks()
+
+	for _, l := range lanes {
+		extraTasks.JoinLane(l)
+	}
+
+	chg := mainTask.Change()
+	if chg != nil {
```

Comment:
do we use it in cases where this is true?  I would think  it's a panic situation if we hit this

Reply:
I was wondering the same - I *guess* (pawel will confirm) that the intend is that the helper can be used in situations when there is no change created yet (when we construct the list of tasksets). But given that we both wondered I think we need a comment here. Also (maybe) YAGNI - unless we actually use this we skip having it?

Reply:
Fair point. My motivation was just to be on a safe side as I wasn't really sure. The uses cases I've right now always have a change. But your comment @mvo5 made me realize it might actually be useful not to require a change under some rare circumstances. Sorry for a little messup in the history as I read your comment after initially removing this code, now reverted it.